candidclient/client.go

```diff
@@ -82,29 +92,49 @@ func New(p NewParams) (*Client, error) {
 	}
 	c.Client.Doer = p.Client
 	c.Client.UnmarshalError = httprequest.ErrorUnmarshaler(new(params.Error))
+	c.useUserID = p.UseUserID
 	return &c, nil
 }
 
 // IdentityFromContext implements identchecker.IdentityClient.IdentityFromContext
 // by returning caveats created by IdentityCaveats.
 func (c *Client) IdentityFromContext(ctx context.Context) (identchecker.Identity, []checkers.Caveat, error) {
+	if c.useUserID {
+		return nil, IdentityUserIDCaveats(c.Client.BaseURL), nil
+	}
 	return nil, IdentityCaveats(c.Client.BaseURL), nil
 }
 
 // DeclaredIdentity implements IdentityClient.DeclaredIdentity.
 // On success, it returns a value that implements Identity as
 // well as identchecker.Identity.
 func (c *Client) DeclaredIdentity(ctx context.Context, declared map[string]string) (identchecker.Identity, error) {
+	if c.useUserID {
+		return c.declaredUserIDIdentity(ctx, declared)
+	}
 	username := declared["username"]
+
 	if username == "" {
 		return nil, errgo.Newf("no declared user name in %q", declared)
 	}
-	return &identity{
+
+	return &usernameIdentity{
 		client:   c,
 		username: username,
 	}, nil
 }
 
+func (c *Client) declaredUserIDIdentity(ctx context.Context, declared map[string]string) (identchecker.Identity, error) {
+	userid := declared["userid"]
```

Comment:
nitpick: make "userid" a const?

Reply:
I've decided against that as this is consistent with "username" version.